admin/client/src/legacy/HtmlEditorField.js

```diff
@@ -1566,4 +1582,274 @@ $.entwine('ss', function($) {
     }
   });
 
+  // this is required because the React version of e.preventDefault() doesn't work
+  // this is to stop React Tabs from navigating the page
+  $('.insert-media-react__dialog-wrapper .nav-link').entwine({
+    onclick: (e) => e.preventDefault(),
+  });
+
+  $('#insert-media-react__dialog-wrapper').entwine({
+    Element: null,
+
+    Data: {},
+
+    onunmatch() {
+      // solves errors given by ReactDOM "no matched root found" error.
+      this._clearModal();
+    },
+
+    _clearModal() {
+      ReactDOM.unmountComponentAtNode(this[0]);
```

Comment:
Can you please run some heap analysis to ensure we don't cause a memory leak with this? Open the modal a few times, navigate to different pages, check if the memory use is roughly the same.